magnum_capi_helm/kubernetes.py

```diff
@@ -28,25 +28,30 @@
 CONF = conf.CONF
 
 
-def file_or_data(obj, file_key):
-    """Returns a path to a file containing the requested data.
+def ensure_file_cert(obj, file_key):
+    """Returns the path and cleanup requirements of cert.
+
+    Returns a tuple containing the path to a file with the requesteddata,
+    and whether the file must be cleaned up.
 
     First check if there is a file path already,
     if the data is there, put it in a file,
-    and return a path to the temp directory
+    remember for cleanup and return the created file.
     """
     if file_key in obj:
-        return obj[file_key]
+        return obj[file_key], False
```

Comment:
I'm not certain I like this tuple return, but maybe it's okay. Feedback welcomed.

I initially made this function part of the class Client and updated `self.tempfiles` directly, but this made testing of the function in isolation difficult.

Reply:
I get what you mean, but this works well enough for me.